internal/loader/configuration_client_manager.go

```diff
@@ -0,0 +1,437 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package loader
+
+import (
+	"azappconfig/provider/internal/properties"
+	"context"
+	"fmt"
+	"math"
+	"net"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	acpv1 "azappconfig/provider/api/v1"
+
+	"math/rand"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azappconfig"
+	"github.com/google/uuid"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+)
+
+//go:generate mockgen -destination=mocks/mock_configuration_client_manager.go -package mocks . ClientManager
+
+type ConfigurationClientManager struct {
+	ReplicaDiscoveryEnabledEnabled bool
+	StaticClientWrappers           []*ConfigurationClientWrapper
+	DynamicClientWrappers          []*ConfigurationClientWrapper
+	validDomain                    string
```

Comment:
what's the guidance of naming? inconsistent here: some fields are camel case, others are pascal case.

Reply:
The type starting with an uppercase letter are exported to other packages. Those starting with a lowercase letter can be used only inside the package.